src/rt-app_parse_config.c

```diff
@@ -325,40 +335,50 @@ static void
 parse_resources(struct json_object *resources, rtapp_options_t *opts)
 {
 	struct lh_entry *entry; char *key; struct json_object *val; int idx;
+	int size;
 
 	log_info(PFX "Parsing resource section");
 
+	/*
+	 * Create at least a "empty" struct that will then be filled either will
+	 * parsing resources table or when parsing task's event
+	 */
+	opts->resources = malloc(sizeof(rtapp_resources_t));
+	opts->resources->nresources = 0;
+
 	if (!resources) {
 		log_info(PFX "No resource section Found");
 		return;
 	}
 
 	if (json_object_is_type(resources, json_type_object)) {
-		opts->nresources = 0;
+		opts->resources->nresources = 0;
 		foreach(resources, entry, key, val, idx) {
-			opts->nresources++;
+			opts->resources->nresources++;
 		}
 
-		log_info(PFX "Found %d Resources", opts->nresources);
-		opts->resources = malloc(sizeof(rtapp_resource_t) * opts->nresources);
```

Comment:
Sigh. I see the code already doesn't do that. Looking at the code we have a few of these actually. Oh well.